src/utils/index.ts

```diff
@@ -39,7 +39,9 @@ export function getErrorMessage(value: unknown): string {
     }
 
     if (isRouteErrorResponse(value)) {
-        return value.data.message;
+        if (typeof value.data === 'string') return value.data;
+        else if (typeof value.data.message === 'string') return value.data.message;
```

Comment:
Let's immediately cast data to unknown to make sure we're not doing something stupid:

```ts
const data: unknown = value.data;
```

And add a comment:
If the thrown error is an instance of `Error`, React Router will convert it to a string. If it's a JSON response like `throw json({ message: "error" })` it will parse that JSON into `ErrorResponse.data`.